### PR TITLE
High Performace Order Storage support for WooCommerce

### PR DESCRIPTION
**Pull request by jnmrclmbsjse**
## What?
[WooCommerce HPOS](https://woocommerce.com/document/high-performance-order-storage/) - Adjustments to support both HPOS and legacy and declare compatibility for it.
## Why?
Order types and order meta data were moved into their own tables, which might result to incompatibility for legacy approaches where we use wpdb or raw queries, or either checking for post types manually.

Screens (its types) and URLs were also changed.
Legacy: 
wp-admin/post.php?action=edit&post={n}
HPOS enabled:
admin.php?page=wc-orders&action=edit&id={n}

## How?
**Order meta data**
All `post_meta` methods changed to `order->{action}_meta_data` to properly handle both legacy and hpos

**Metabox**
Registration of metabox hooks now supports HPOS by checking relevant classes and enable state, but kept the legacy handling.
Force sync (action) adjustment as `post_action_{action}` is no longer working if hpos is enabled
Global `$theorder` is `WP_Post` when hpos is disabled, `WC_Order` otherwise, so adjustments were made for nonce.
Generation of edit url is adjusted.

**Admin notices**
Usage of OrderUtil to check if post is an order, and wc_get_order_types instead of hardcoded `shop_order` and `shop_order_placehold` as fallback.

**Plugin**
Declared compatibility to HPOS
Added tested up to as WooCommerce reads it to check for plugins that uses WooCommerce

## Testing?
`wp-admin/admin.php?page=wc-settings&tab=advanced&section=features`

_When saved as wordpress post (legacy) is enabled_
_Given that Enable compatibility mode is enabled_
_Then orders and metadata will be saved in both post and wc order tables_

_When HPOS is enabled_
_Given that Enable compatibility mode is enabled_
_Then orders and metadata will be saved in both post and wc order tables_

_When saved as wordpress post (legacy) is enabled_
_Given that Enable compatibility mode is disabled_
_Given a new order was created_
_Then orders and metadata will be saved as post and HPOS can't be enabled until all orders are synced_

_When HPOS is enabled_
_Given that Enable compatibility mode is disabled_
_Given a new order was created_
_Then orders and metadata will be saved in new orders tables and legacy can't be enabled until all orders are synced_

## Anything Else?
Enable compatibility mode should create a scheduler to sync orders on both ends.
Removed deprecated `OrderHandler::delete` as it uses the legacy approach and no longer used by our plugin.
Order metabox is shown 3 metaboxes below if hpos is enabled. Tried changing priorities but can't sort.

### CODE REVIEW

File: src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php [hunk 1]
```diff
@@ -2,22 +2,65 @@
...
+
+    /**
+     * @deprecated
+     * Support for old WooCommerce which does not support High Performance Order Storage
+     * To be removed in future
+     */
+    private function registerLegacyOrderScreen(): void
+    {
+        // Support for old WooCommerce which does not support High Performance Order Storage
+        // Remove in future
```
Comment (ikulis) [inline src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php:60]:
```suggestion
```

do not leave hidden todos :P

Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Backoffice/MetaBoxes/OrderSyncMetaBox.php]: Removed.

File: src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php [hunk 2]
```diff
@@ -367,8 +383,11 @@ protected function processItem($order): void
 
         if ($order->meta_exists(OrderHandler::TO_BE_SYNCHRONIZED_META_KEY)) {
             $order->delete_meta_data(OrderHandler::TO_BE_SYNCHRONIZED_META_KEY);
-            $order->save();
         }
+
+        $order->save();
+        // Remove ignore
+        $storekeeper_ignore_order_id = null;
```
[the remark this answers is not in the document]
Comment (ikulis) [reply src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php:390]: image

**Review by ikulis — APPROVED** [review]

File: src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (jnmrclmbsjse) [reply src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php:390]: image
It is actually triggered by these.
So updating the meta data will retrigger the updateWithIgnore

**Pull request merged**